src/cargo/sources/git/utils.rs

```diff
@@ -31,60 +31,89 @@ where
     s.collect_str(t)
 }
 
+/// A short abbreviated OID.
+///
+/// Exists for avoiding extra allocations in [`GitDatabase::to_short_id`].
+// TODO(refactor): maybe we don't need this abstraction.
```

Comment:
What are the plans for all of these TODOs?  They read like personal notes of ideas you have to improve things.  Personally, I prefer not to commit TODOs because they quickly get stale and and serve as a second backlog of information.  Sometimes, I rework TODOs into explanations (e.g. the `location` one being "this is similar to `GitDatabase::path`)

Reply:
Valid. I'll keep them in my personal branch.